src/uu/split/src/number.rs

```diff
@@ -0,0 +1,513 @@
+/// 'b', 'c', etc.) and concatenated with a prefix of `n - 2`
+/// instances of the character 'z'.
+///
+/// This notion of displaying the number is specific to the `split`
+/// program.
+#[derive(Clone)]
+pub struct DynamicWidthNumber {
+    radix: u8,
+    digits: Vec<u8>,
+}
+
+impl DynamicWidthNumber {
+    /// Instantiate a number of the given radix, starting with width 2.
+    ///
+    /// This associated function returns a new instance of the struct
+    /// with the given radix and a width of two digits, both 0.
+    pub fn new(radix: u8) -> DynamicWidthNumber {
+        DynamicWidthNumber {
+            radix,
+            digits: vec![0, 0],
+        }
+    }
+
+    /// Set all digits to zero.
+    fn reset(&mut self) {
+        for i in 0..self.digits.len() {
+            self.digits[i] = 0;
+        }
+    }
+
+    /// Increment this number.
+    ///
+    /// This method adds one to this number. The first time that the
+    /// most significant digit would achieve its highest possible
+    /// value (that is, `radix - 1`), then all the digits get reset to
+    /// 0 and the number of digits increases by one.
+    ///
+    /// This method never returns an error.
+    fn increment(&mut self) -> Result<(), Overflow> {
+        for i in (0..self.digits.len()).rev() {
+            // Increment the current digit.
+            self.digits[i] += 1;
+
+            // If the digit overflows, then set it to 0 and continue
+            // to the next iteration to increment the next most
+            // significant digit. Otherwise, terminate the loop, since
+            // there will be no further changes to any higher order
+            // digits.
+            if self.digits[i] == self.radix {
+                self.digits[i] = 0;
+            } else {
+                break;
+            }
+        }
+
+        // If the most significant digit is at its maximum value, then
+        // add another digit and reset all digits zero.
+        if self.digits[0] == self.radix - 1 {
+            self.digits.push(0);
+            self.reset();
+        }
+        Ok(())
+    }
+}
+
+impl Display for DynamicWidthNumber {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self.radix {
+            10 => {
+                let num_fill_chars = self.digits.len() - 2;
+                let digits: String = self.digits.iter().map(|d| (b'0' + d) as char).collect();
+                write!(
+                    f,
+                    "{empty:9<num_fill_chars$}{digits}",
+                    empty = "",
+                    num_fill_chars = num_fill_chars,
+                    digits = digits,
+                )
+            }
+            26 => {
+                let num_fill_chars = self.digits.len() - 2;
+                let digits: String = self.digits.iter().map(|d| (b'a' + d) as char).collect();
+                write!(
+                    f,
+                    "{empty:z<num_fill_chars$}{digits}",
```

Comment:
For example, the number whose digits are `vec![1, 2, 3]` gets displayed as `"zbcd"`.